cachito/workers/scm.py

```diff
@@ -118,10 +124,19 @@ def update_and_archive(self, previous_archive):
                 raise CachitoError("Failed to fetch from the remote Git repository")
 
             self._reset_git_head(repo)
+            if gitsubmodule:
+                self.update_git_submodule(repo)
+
             self._create_archive(repo.working_dir)
 
-    def fetch_source(self):
-        """Fetch the repo, create a compressed tar file, and put it in long-term storage."""
+    def fetch_source(self, gitsubmodule=False):
+        """Fetch the repo, create a compressed tar file, and put it in long-term storage.
+
+        :param bool gitsubmodule: a bool to determine whether git submodules need to be processed.
+        """
+        if gitsubmodule:
+            self.sources_dir = SourcesDir(self.repo_name, f"{self.ref}-with-submodule")
```

Comment:
(optional): Should this be `-with-submodules` (plural) since we expect there could be more than one?